### ISSUE TITLE
Wrong setting of endOffset while using dataProcessingOption in agent.json file

### ISSUE THREAD

**Issue by kiranclarivate** [awslabs/amazon-kinesis-agent#116]
Hi,

I could see that in agent endOffset is considered for the converted data.
e.g.
log file looks like  : - 
Word
As

agent.json: - 
{
        "cloudwatch.emitMetrics": true,
        "cloudwatch.endpoint": "monitoring.us-west-2.amazonaws.com",
        "log.level":"TRACE",
        "checkpointFile": "/tmp/aws-kinesis-agent-checkpoints/main.log",
        "kinesis.endpoint": "kinesis.us-west-2.amazonaws.com",
        "firehose.endpoint": "firehose.us-west-2.amazonaws.com",
        "awsAccessKeyId": "",
        "awsSecretAccessKey": "",
        "flows": [{
                "filePattern": "/tmp/logs/e2e/cps/*.txt",
                "deliveryStream": "",
                "initialPosition": "START_OF_FILE",
                "dataProcessingOptions": [{
                        "optionName": "LOGTOJSON",
                        "logFormat": "COMMONAPACHELOG",
                        "customFieldNames": ["Word"],
                        "matchPattern": "([a-zA-Z]*)$"
                }]
        }]
}

With dataProcessingOption ,agent will convert this data ,which looks like {"Word":"As"}\n
In checkpoint file, endoffset will be set to 14 considering all the curly braces,double quotes,etc.
Once we stop and start the engine,ideally agent should consider this log file to be already read and shouldnt send the data.Instead it throws this warning : - 
We get Checkpoint offset (i.e.14) is beyound the end of the current file. (i.e. Size of file = 8).This suggests that we failed to track the correct file referenced in the checkpoint.

And agent will start reading the log file again and send the data which has already been sent.

Thanks,
Kiran

**Comment by kiranclarivate** [comment]
The above mentioned analysis is the root cause for #114

**Comment by kiranclarivate** [comment]
We have received a fix for this error in Release 1.1.3.But now we are facing same issues with checkpoint, if the file has header.

**Comment by chaochenq** [comment]
@kiranclarivate can you elaborate? did the fix actually fix your issue? Or does it only occur when you have the header configured now?

**Comment by kiranclarivate** [comment]
@chaochenq  ,
Yes the fix fixed the issue .It occurs now when the file has header.

Thanks,
Kiran
